File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string line) {
        // return the message
        int firstspace = line.find(" ");
        return line.substr(firstspace+1);
    }

    std::string log_level(std::string line) {
        // return the log level
        int endLl = line.find(":");
        return line.substr(1,endLl-2);

    }

    std::string reformat(std::string line) {
        // return the reformatted message
        return log_line::message(line) + " (" + log_line::log_level(line) + ")";
    }
}

File: solutions/julia/log-levels/1/log-levels.jl
function message(msg)
	split_msg = split(msg)
	return join(split_msg[2:length(split_msg)], " ")
end

function log_level(msg)
	split_msg = split(msg)
	return lowercase(split_msg[1][2:length(split_msg[1]) - 2])
end

function reformat(msg)
	string(message(msg), " (", log_level(msg), ")")
end
